src/features/posts/postSlice.ts

```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
```

Comment:
Ensure that the `Post` type is correctly defined and imported from the `../../types/Post` file. Verify its existence and correctness.